NeuroID/NeuroIDClass/Extensions/NIDUser.swift

```diff
@@ -6,84 +6,78 @@
-            }
-            CURRENT_ORIGIN_CODE = SessionOrigin.NID_ORIGIN_CODE_FAIL.rawValue
-
-            sendOriginEvent(origin: CURRENT_ORIGIN!, originCode: CURRENT_ORIGIN_CODE!, originSessionID: userId)
             return false
         }
-
         return true
     }
-
-    internal static func setGenericUserID(userId: String, type: UserIDTypes, completion: (Bool) -> Bool) -> Bool {
-        let validID = NeuroID.validateUserID(userId)
-        if !validID { return completion(validID) }
-
-        NIDLog.d(tag: "\(type)", "\(userId)")
-
+    
+    internal static func setGenericUserID(type: UserIDTypes, genericUserID: String, userGenerated: Bool = true ) -> Bool{
+        let validID = validateUserID(genericUserID)
+        
+        let originRes = getOriginResult(idValue: genericUserID, validID: validID, userGenerated: userGenerated, idType: type)
+        sendOriginEvent(originResult: originRes)
+        
+        if(!validID){return false}
+        
+        NIDLog.d(tag: "\(type)", "\(genericUserID)")
+        //    Queue user id event to be sent
         let setUserEvent = NIDEvent(
             sessionEvent: type == .userID ?
-                NIDSessionEventName.setUserId : NIDSessionEventName.setRegisteredUserId
+            NIDSessionEventName.setUserId : NIDSessionEventName.setRegisteredUserId
         )
-
-        setUserEvent.uid = userId
-
+        
+        setUserEvent.uid = genericUserID
+        
         if !NeuroID.isSDKStarted {
             saveQueuedEventToLocalDataStore(setUserEvent)
         } else {
             saveEventToLocalDataStore(setUserEvent)
         }
-
-        return completion(true)
+        return true
     }
-
-    internal static func sendOriginEvent(origin: String, originCode: String, originSessionID: String) {
+    
+    internal static func sendOriginEvent(originResult: SessionIDOriginalResult){
         let sessionIdCodeEvent =
-            NIDEvent(
-                sessionEvent: NIDSessionEventName.setVariable,
-                key: "sessionIdCode",
-                v: originCode
-            )
-
+        NIDEvent(
+            sessionEvent: NIDSessionEventName.setVariable,
+            key: "sessionIdCode",
+            v: originResult.originCode
+        )
+        
         let sessionIdSourceEvent =
-            NIDEvent(
-                sessionEvent: NIDSessionEventName.setVariable,
-                key: "sessionIdSource",
-                v: origin
-            )
-
+        NIDEvent(
+            sessionEvent: NIDSessionEventName.setVariable,
+            key: "sessionIdSource",
+            v: originResult.origin
+        )
+        
         let sessionIdEvent =
-            NIDEvent(
-                sessionEvent: NIDSessionEventName.setVariable,
-                key: "sessionId",
-                v: originSessionID
-            )
-
+        NIDEvent(
+            sessionEvent: NIDSessionEventName.setVariable,
+            key: "sessionId",
+            v: originResult.idValue
```

Comment:
Can we wrap this is '' to clearly indicate if they set an empty id?

Additionally can we set an event for the idType?